gadgets/trace_exec/program.bpf.c

```diff
@@ -218,36 +181,20 @@ static __always_inline bool has_upper_layer()
 SEC("tracepoint/syscalls/sys_exit_execve")
 int ig_execve_x(struct syscall_trace_exit *ctx)
 {
-	u64 id;
-	pid_t pid, tgid;
-	pid_t execs_lookup_key;
-	u64 *pid_sum;
 	int ret;
 	struct event *event;
+	pid_t execs_lookup_key;
 	u32 uid = (u32)bpf_get_current_uid_gid();
 
 	if (valid_uid(targ_uid) && targ_uid != uid)
 		return 0;
-	id = bpf_get_current_pid_tgid();
-	pid = (pid_t)id;
-	tgid = id >> 32;
-	ret = ctx->ret;
 
-	pid_sum = bpf_map_lookup_elem(&pid_by_tgid, &tgid);
-	if (!pid_sum)
+	ret = ctx->ret;
+	execs_lookup_key = gadget_get_exec_caller_pid(ret);
```

Comment:
I had a look at the uid filter:
```c
if (valid_uid(targ_uid) && targ_uid != uid)
		return 0;
```
We use this filter both in `tracepoint/syscalls/sys_enter_execve` and `tracepoint/syscalls/sys_exit_execve`. If the uid changes in between, `gadget_get_exec_caller_pid` will not clean the map properly.

I tried with:
```
sudo -E ig run ghcr.io/inspektor-gadget/gadget/trace_exec:latest --verify-image=false --host --uid 1000
```
And run `/bin/mount` as regular user 1000. `/bin/mount` is a setuid executable, so I was expecting to trigger a bug.
However `bpf_get_current_uid_gid` is returning the _real_ credentials and not the _effective_ credentials. Setuid executables only change the _effective_ credentials, so no bug were triggered. Applications have to call `setreuid` to change the _real_ credentials. _Real_ credentials cannot change in the middle of an execve because on Linux credentials are a property of threads and not processes.

The reason why it works is complicated, so I'm thinking the filter should be removed in the `tracepoint/syscalls/sys_exit_execve`, or at least move `gadget_get_exec_caller_pid` before the filter.

Reply:
Opened #3029